test/e2e/utils/kubectl.go

```diff
@@ -44,7 +44,7 @@ func (k *Kubectl) CommandWithInput(stdinInput string, cmdOptions ...string) (str
 	}
 	go func() {
 		defer stdin.Close()
-		io.WriteString(stdin, stdinInput)
+		_, err = io.WriteString(stdin, stdinInput)
```

Comment:
You are not using `err` so you should use `_` instead.

Reply:
The error is used in the return.

Reply:
Wait, you are using the `err` variable from outside the goroutine. That's even more danagerous (race conditions) and you won't know which `err` (the one returned by `io.WriteString(stdin, stdinInput)` or the one returned by `k.Run(cmd)`) you are returning. Maybe the first fails and the second doesn't and a real error is overriden by a `nil`.

Reply:
This doesn't solve the issue as it is running in a goroutine.